app/controllers/albums_controller.rb

```diff
@@ -19,41 +19,43 @@ def new
   def edit
   end
 
-  # POST /albums or /albums.json
+  # POST /albums
   def create
     @album = Album.new(album_params)
 
-    respond_to do |format|
-      if @album.save
-        format.html { redirect_to @album, notice: "Album was successfully created." }
-        format.json { render :show, status: :created, location: @album }
-      else
-        format.html { render :new, status: :unprocessable_entity }
-        format.json { render json: @album.errors, status: :unprocessable_entity }
-      end
+    if @album.save
+      redirect_to @album, notice: "Album was successfully created."
+    else
+      render :new, status: :unprocessable_entity
     end
   end
 
-  # PATCH/PUT /albums/1 or /albums/1.json
+  # PATCH/PUT /albums/1
   def update
-    respond_to do |format|
-      if @album.update(album_params)
-        format.html { redirect_to @album, notice: "Album was successfully updated." }
-        format.json { render :show, status: :ok, location: @album }
-      else
-        format.html { render :edit, status: :unprocessable_entity }
-        format.json { render json: @album.errors, status: :unprocessable_entity }
+    if @album.update(album_params_no_images)
+      if params[:album][:images].present?
```

Comment:
I think attaching the optionally provided images is an important concept in the domain, so it deserves an own name and method.

In my opinion an easily readable solution would comform the folowing api:

```suggestion
    images = params[:album][:images]
    @album.attach_images(images) if images.present?
```

Reply:
My main concern is this is too much logic in a controller and we handle the same thing(uploding images) in two different ways.